### PR TITLE
[update] Added telegraf to how-to

### PR DESCRIPTION
**Pull request by Rajakavitha1**
# Description

- Updated the how-to to ingest data using telegraf
- Updated the index page to install and use timescaledb-parallel-copy

PS: Reviewers after this how to is merged may be we can remove Telegraf from the tutorials. With redirects to the new how-to

# Links

Fixes #[insert issue link, if any]

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 1]
```diff
@@ -1,24 +1,215 @@
...
+
+#### Configuring the PostgreSQL output plugin
+
+1.  Open the `telegraf.conf` file using an editor of your choice:
+
+    ```bash
+    nano telegraf.conf
+    ```
+
+1.  Set the `connection` parameter in the `[[outputs.postgresql]] section to
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:84]:
```suggestion
1.  Set the `connection` parameter in the `[[outputs.postgresql]]` section to
```

File: timescaledb/how-to-guides/ingest-data/ingest-parallel-copy.md [hunk 2]
```diff
@@ -6,15 +6,9 @@ tags: [insert]
...
 # TimescaleDB parallel copy
-For bulk inserting historical data, you can use the TimescaleDB parallel copy
-tool, called `timescaledb-parallel-copy`. Install the tool from the TimescaleDB
-repository with this command:
 
-```bash
-go get github.com/timescale/timescaledb-parallel-copy/cmd/timescaledb-parallel-copy
-```
+To insert bulk historical data, you can use the TimescaleDB parallel copy tool.
+For more information about installing and using `timescale-parallel-copy` tool,
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-parallel-copy.md:11]:
```suggestion
For more information about installing and using the `timescale-parallel-copy` tool,
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 3]
```diff
@@ -1,24 +1,215 @@
...
+Telegraf is an open source tool. It contains over 200 plugins for gathering and
+writing different types of data written by people who work with that data.
 
-Timescale have an unofficial build of Telegraf version 1.13.0 with the plugin
-added. For more information about installing the Timescale Telegraf binaries
-with the plugin, see the [telegraf-tutorial][telegraf-tutorial].
+## Before you begin
 
-[telegraf-tutorial]: /timescaledb/:currentVersion:/tutorials/telegraf-output-plugin/
+*   [Install Telegraf][install-telegraf] on the system you want to collect metrics.
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:22]:
```suggestion
*   [Install Telegraf][install-telegraf] on the system where you want to collect metrics.
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 4]
```diff
@@ -1,24 +1,215 @@
...
+*   Install TimescaleDB or create a [TimescaleDB service][create-service] on
+    Timescale Cloud.
+*   Gather the connection details for [TimescaleDB][connect-timescaledb].
+*   Install Grafana or create a Grafana service on [Managed Service for TimescaleDB][grafana-mst]
+
+### Ingesting metrics using Telegraf plugin to TimescaleDB
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:28]:
```suggestion
## Ingest metrics using Telegraf plugin to TimescaleDB
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 5]
```diff
@@ -1,24 +1,215 @@
...
+
+<procedure>
+
+#### Generating a configuration file
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:32]:
```suggestion
### Generating a configuration file
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 6]
```diff
@@ -1,24 +1,215 @@
...
+
+1.  Check the version of Telegraf that you installed:
+
+    ```bash
+    telegrapf --version
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:37]:
```suggestion
    telegraf --version
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 7]
```diff
@@ -1,24 +1,215 @@
...
+    ```
+
+1.  Generate a sample configuration file for Telegraf using:
+
+    ```bash
+    telegraf --input-filter=cpu --output-filter=postgresql config > telegraf.conf
+    ```
+
+    A configuration file enables a CPU input plugin that samples various metrics
+    of CPU usage, and the PostgreSQL output plugin. The file also includes all
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:47]:
```suggestion
    about CPU usage, and the PostgreSQL output plugin. The file also includes all
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 8]
```diff
@@ -1,24 +1,215 @@
...
+    available input, output, processor, and aggregator plugins, commented out
+    and you can enable them as required.
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:49]:
```suggestion
    available input, output, processor, and aggregator plugins. These are commented out
    commented out by default. You can enable them as required.
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 9]
```diff
@@ -1,24 +1,215 @@
...
+    > cpu,cpu=cpu0,host=localhost usage_guest=0,usage_guest_nice=0,usage_idle=90.00000000087311,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=6.000000000040018,usage_user=3.999999999996362 1669640025000000000
+    > cpu,cpu=cpu1,host=localhost usage_guest=0,usage_guest_nice=0,usage_idle=92.15686274495818,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=5.882352941192206,usage_user=1.9607843136712912 1669640025000000000
+    > cpu,cpu=cpu2,host=localhost usage_guest=0,usage_guest_nice=0,usage_idle=91.99999999982538,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=3.999999999996362,usage_user=3.999999999996362 1669640025000000000
+    ```
+
+</procedure>
+
+<procedure>
+
+#### Configuring the PostgreSQL output plugin
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:76]:
```suggestion
### Configuring the PostgreSQL output plugin
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 10]
```diff
@@ -1,24 +1,215 @@
...
+    # ]
+    #  table_template=`CREATE TABLE IF NOT EXISTS {TABLE}({COLUMNS}); SELECT create_hypertable({TABLELITERAL},'time',chunk_time_interval := INTERVAL '1 week',if_not_exists := true);`
+
+    ```
+
+</procedure>
+
+<procedure>
+
+#### Viewing the metrics collected by Telegraf in TimescaleDB
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:107]:
```suggestion
### Viewing the metrics collected by Telegraf in TimescaleDB
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 11]
```diff
@@ -1,24 +1,215 @@
...
+
+1.  Run Telegraf to collect the metrics:
+
+    ```bash
+    telegraf --config telegraf.conf
+    ```
+
+    The output uses loaded inputs `cpu`and outputs `postgresql` along with
+    `global tags`, and the intervals with which the agent collects the data from
+    the inputs, and flush to the outputs.
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:117]:
```suggestion
    the inputs, and flushes to the outputs.
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 12]
```diff
@@ -1,24 +1,215 @@
...
+      cpu1      |  4.9013756546309155
+      cpu6      |  0.6719913538159535
+      cpu5      |  1.0512637475474937
+      cpu3      |   3.871919066617788
+    ```  
+
+</procedure>
+
+For more information about the options that you can configure in Telegraf,
+see [PostgreQL output plugin] [output-plugin]
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:184]:
```suggestion
see [PostgreQL output plugin][output-plugin].
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 13]
```diff
@@ -1,24 +1,215 @@
...
+
+<procedure>
+
+#### Visualizing data in Grafana
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:188]:
```suggestion
### Visualizing data in Grafana
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 14]
```diff
@@ -1,24 +1,215 @@
...
+
+1.  Log in to Grafana and navigate to `Configuration` → `Data sources`. The data
+    sources page lists previously configured data sources for the Grafana
+    instance.
+1.  Click `Add data source` to see a list of all supported data sources.
+1.  Type `PostgreSQL` in the search field and click `Select`.
+1.  Configure the data source:
+    *   In the `Name` field, type name that you would like for dataset on TimescaleDB.
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:196]:
```suggestion
    *   In the `Name` field, type the name that you would like to use for the dataset.
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 15]
```diff
@@ -1,24 +1,215 @@
...
+    *   In the `PostgreSQL Connection` section, type the  `Database`, `User`,
+        and `Password` fields using the `.sql` file that you downloaded when
+        creating the TimescaleDB service. 
+    *   In the `Host` type `<HOST>:<PORT>` from the `.sql` file that you downloaded.
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:200]:
```suggestion
    *   In the `Host` field, type `<HOST>:<PORT>` from the `.sql` file that you downloaded.
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 16]
```diff
@@ -1,24 +1,215 @@
...
+    *   Set `TLS/SSL Mode` as `require`.
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:201]:
```suggestion
    *   Set `TLS/SSL Mode` to `require`.
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 17]
```diff
@@ -1,24 +1,215 @@
...
+    *   In `PostgreSQL details` enable `TimescaleDB`
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:202]:
```suggestion
    *   In `PostgreSQL details`, enable `TimescaleDB`
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 18]
```diff
@@ -1,24 +1,215 @@
...
+1.  Click `Save & test` button. If the connection is successful
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:203]:
```suggestion
1.  Click `Save & test`. If the connection is successful,
```

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 19]
```diff
@@ -1,24 +1,215 @@
...
+
+</procedure>
+
+When you have configured TimescaleDB as a data source in Grafana, you can create
+panels that are populated with data using SQL.
+
+[output-plugin]: https://github.com/influxdata/telegraf/blob/release-1.24/plugins/outputs/postgresql/README.md
+[install-telegraf]: https://docs.influxdata.com/telegraf/v1.21/introduction/installation/
+[create-service]: /install/latest/installation-cloud/
+[connect-timescaledb]: /timescaledb/latest/how-to-guides/connecting/about-connecting/
```
Comment (charislam) [inline timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:214]:
```suggestion
[connect-timescaledb]: /timescaledb/:currentVersion:/how-to-guides/connecting/about-connecting/
```

**Review by charislam — APPROVED** [review]

File: timescaledb/how-to-guides/ingest-data/ingest-telegraf.md [hunk 20]
```diff
@@ -200,7 +200,7 @@ see [PostgreQL output plugin] [output-plugin]
     *   In the `Host` field, type `<HOST>:<PORT>` from the `.sql` file that you downloaded.
     *   Set `TLS/SSL Mode` to `require`.
     *   In `PostgreSQL details`, enable `TimescaleDB`
-1.  Click `Save & test` button. If the connection is successful
+1.  Click `Save & test`. If the connection is successful,
     `Database Connection OK` appears.
```
[the remark this answers is not in the document]
Comment (Loquacity) [reply timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:204]: "appears" always feels a little magical to me, somehow 🪄 ✨

[the remark this answers is not in the document]
Comment (Rajakavitha1) [reply timescaledb/how-to-guides/ingest-data/ingest-telegraf.md:204]: to me as well :)

**Pull request merged**